### ISSUE TITLE
Can't get action to open PR: instead tries to push directly and fails. (Protected branch)

### ISSUE THREAD

**Issue by Cameronsplaze** [peter-evans/create-pull-request#780]
### Subject of the issue

I'm trying to open a PR to a protected branch, and the action fails due to not being able to push. I'm confused on why it's trying to push directly instead of open a PR, but I have a weird use case that might be causing this. I'm trying to keep branch develop and main in sync, such that when someone opens a PR against main, this action will open a new one against develop, and only let main be able to merge when develop finishes it's suite of tests.

### Steps to reproduce
```yaml
  automerge-prod:
    name: Automerge [prod]
    if: >
      github.event.pull_request.base.ref == 'main'
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          ref: ${{ github.event.pull_request.head.ref }}

      - name: "Open PR to develop"
        uses: peter-evans/create-pull-request@v3.8.2
        with:
          branch: ${{ github.event.pull_request.head.ref }}
          base: develop
          title: "Update from PR on main branch"
          labels: 'auto merge'
```
TLDR: Here, ${{ github.event.pull_request.head.ref }} isn't develop or main, but some changes on a different/temp branch. I want this action to run on main, but open a PR from ${{ github.event.pull_request.head.ref }} to develop. Is this possible?

**Comment by peter-evans** [comment]
Hi @Cameronsplaze 

Your use case is not the default behaviour of this action, but it can be done. Please see this example: https://github.com/peter-evans/create-pull-request/blob/master/docs/examples.md#keep-a-branch-up-to-date-with-another

**Comment by Cameronsplaze** [comment]
Hey Peter!

Sorry it's taken a bit to get back to you. I was hitting a weird bug that I thought had to do with this, and I wanted to troubleshoot it. Turns out it was something else.
That link definitely got me past my problem, thank you for that!
I was thrown off a little by the last line in it's example, opening the PR to "production-promotion", instead of just production. There's probably workflows where that makes sense though.

Here's what I finally ended up with, if anyone else stumbles across this:
```yaml

      - uses: actions/checkout@v2
        with:
          ref: develop

     # Reset from wherever the PR came from
      - name: Reset branch to ${head_of_pr}
        run: |
          git fetch origin ${head_of_pr}:${head_of_pr}
          git reset --hard ${head_of_pr}
        env:
          head_of_pr: ${{ github.event.pull_request.head.ref }}

      - name: "Open PR to develop"
        uses: peter-evans/create-pull-request@v3.8.2
        with:
          # Let bot open the PR, so auto merge can trigger
          token: ${{ secrets.BOT_TOKEN }}
          labels: 'auto merge, dependancies'
          # Open from wherever the PR was from, to develop:
          branch: ${{ github.event.pull_request.head.ref }}
          base: develop
          title: "Automatic merge triggered from ${{ github.event.pull_request.head.ref }} to develop"
```

**Comment by altjx** [comment]
> I was thrown off a little by the last line in it's example, opening the PR to "production-promotion", instead of just production. 

Same here. Wow that was a PITA trying to wrap my head around that. Thanks for your example and glad I'm not the only one.
